File: app/src/main/java/com/proxy/shadowsocksr/impl/plugin/proto/ProtocolChooser.java
package com.proxy.shadowsocksr.impl.plugin.proto;

import java.util.HashMap;

public final class ProtocolChooser
{
    public static AbsProtocol getProtocol(String protoName, String rmtIP, int rmtPort, int tcpMss,
            HashMap<String, Object> shareParam)
    {
        switch (protoName)
        {
        case "verify_simple":
            return new VerifySimpleProtocol(rmtIP, rmtPort, tcpMss,shareParam);
        case "auth_simple":
            return new AuthSimpleProtocol(rmtIP, rmtPort, tcpMss, shareParam);
        case "plain":
        default:
            return new SSProtocol(rmtIP, rmtPort, tcpMss,shareParam);
        }
    }
}

File: app/src/main/java/com/proxy/shadowsocksr/impl/plugin/proto/ProtocolChooser.kt
package com.proxy.shadowsocksr.impl.plugin.proto

import java.util.HashMap

class ProtocolChooser
{
    companion object
    {
        fun getProtocol(protoName: String, rmtIP: String, rmtPort: Int, tcpMss: Int,
                        shareParam: HashMap<String, Any>): AbsProtocol
        {
            when (protoName)
            {
                "verify_simple" -> return VerifySimpleProtocol(rmtIP, rmtPort, tcpMss, shareParam)
                "auth_simple" -> return AuthSimpleProtocol(rmtIP, rmtPort, tcpMss, shareParam)
            //... -> ...
                else -> return SSProtocol(rmtIP, rmtPort, tcpMss, shareParam)//default plain
            }
        }
    }
}
